### PR TITLE
CON-3172 Fixing Credit Scoring project

### PR DESCRIPTION
**Pull request by Oumaimafisaoui**
Resolves CON-3172

Changes Overview:

- Reorganized project structure for better modularity and clarity
- Split Jupyter notebook into separate Python scripts
- Enhanced documentation throughout the project

Testing:

- Ran all scripts to ensure they execute without errors
- Verified model performance meets or exceeds previous benchmarks
- Checked that all required outputs are generated correctly

### CODE REVIEW

**Review by MSilva95 — CHANGES_REQUESTED** [review]
There is quite some stuff to change, and I saw some more that I would prefer to do the change myself or speak to you directly so that I can explain better

File: subjects/ai/credit-scoring/README.md [hunk 1]
```diff
@@ -1,27 +1,31 @@
 ## Credit scoring
 
+### 1. Introduction :
```
[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md]: fixed

File: subjects/ai/credit-scoring/README.md [hunk 2]
```diff
@@ -94,16 +98,16 @@ project
 ```
 
 - `README.md` introduces the project and shows the username.
-- `environment.yml` contains all libraries required to run the code.
+- `requirements.txt` contains all libraries required to run the code.
```
[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md:112]: fixed

File: subjects/ai/credit-scoring/README.md [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
 
-### Resources
+The ability to understand the underlying factors of credit scoring is important. Credit scoring is subject to more and more regulation, so transparency is key. And more generally, more and more companies prefer transparency to black box models.
 
 Historical timeline of machine learning techniques applied to credit scoring
 
 - [Machine Learning or Econometrics for Credit Scoring: Let’s Get the Best of Both Worlds](https://hal.archives-ouvertes.fr/hal-02507499v3/document)
 
-### Scoring model
+#### a - Scoring model
```
[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md]: fixed

File: subjects/ai/credit-scoring/README.md [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
 
 There are 3 expected deliverables associated with the scoring model:
 
 - An exploratory data analysis notebook that describes the insights you find out in the data set.
 - The trained machine learning model with the features engineering pipeline:
 
   - Do not forget: **Coming up with features is difficult, time-consuming, requires expert knowledge. ‘Applied machine learning’ is basically feature engineering.**
-  - The model is validated if the **AUC on the test set is higher than 75%**.
+  - The model is validated if the **AUC on the test set is higher than 50%**.
```
[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md:29]: fixed

[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md:29]: fixed###

File: subjects/ai/credit-scoring/README.md [hunk 5]
```diff
@@ -55,16 +59,16 @@ Choose the 3 clients of your choice, compute the score, run the visualizations o
   - 1 on which the model is correct and the other on which the model is wrong. Try to understand why the model got wrong on this client.
 - Take 1 client from the test set
 
-### Optional
+#### d - Optional
```
[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md]: fixed

File: subjects/ai/credit-scoring/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oumaimafisaoui) [reply subjects/ai/credit-scoring/README.md]: fixed

**Pull request closed without merging**